addons/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/handler/AutelisHandler.java

```diff
@@ -0,0 +1,401 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.autelis.handler;
+
+import java.io.StringReader;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.apache.commons.httpclient.Credentials;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.httpclient.UsernamePasswordCredentials;
+import org.apache.commons.httpclient.auth.AuthScope;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.autelis.config.AutelisConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+
+/**
+ * 
+ * Autelis Pool Control Binding
+ * 
+ * Autelis controllers allow remote access to many common pool systems. This
+ * binding allows openHAB to both monitor and control a pool system through these 
+ * controllers.
+ * 
+ * @see <a href="http://Autelis.com">http://autelis.com</a>
+ * 
+ * The {@link AutelisHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ * 
+ * @author Dan Cunningham - Initial contribution
+ */
+public class AutelisHandler extends BaseThingHandler {
+
+	private Logger logger = LoggerFactory.getLogger(AutelisHandler.class);
+
+	/**
+	 * Default port to use for connections to a Autelis controller
+	 */
+	private static final int DEFAULT_PORT = 80;
+
+	/**
+	 * Default timeout for http connections to a Autelis controller
+	 */
+	static final int TIMEOUT = 5000;
+	
+	/**
+	 * Autelis controllers will not update their XML immediately after we change a value. To compensate
+	 * we cache previous values for a {@link Item} using the item name as a key.  After a polling
+	 * run has been executed we only update an item if the value is different then what's in the
+	 * cache.  This cache is cleared after a fixed time period when commands are sent.
+	 */
+	private Map<String, State> stateMap = Collections.synchronizedMap(new HashMap<String, State>());
```

Comment:
Ok, I see your point. Give me until tonight to review this part of the code then